tests/provision/provisioning_rules.go

```diff
@@ -0,0 +1,1546 @@
+package tests
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	mediaSsd  = "SSD"
+	mediaSata = "SATA"
+	mediaType = "media_type"
+)
+
+type labelDict map[string]interface{}
+
+//VpsTemplate provides interface for running individual testcases
+type VpsTemplate interface {
+	// Node label and whether it needs to be set on node remove
+	GetLabels() ([]labelDict, int)
+	// Get StorageClass placement_strategy
+	GetScStrategyMap() map[string]string
+	// Vps Spec
+	GetSpec() string
+	// Clean up
+	CleanVps()
+}
+
+var (
+	vpsRulesReplica      = make(map[string]VpsTemplate)
+	vpsRulesVolume       = make(map[string]VpsTemplate)
+	vpsRulesMix          = make(map[string]VpsTemplate)
+	vpsRulesMixScale     = make(map[string]VpsTemplate)
+	vpsRulesPending      = make(map[string]VpsTemplate)
+	vpsRulesDefaultLabel = make(map[string]VpsTemplate)
+)
+
+// Register registers the given vps rule
+func Register(name string, d VpsTemplate, cat int) error {
+
+	if cat == 1 {
+		if _, ok := vpsRulesReplica[name]; !ok {
+			vpsRulesReplica[name] = d
+		} else {
+			return fmt.Errorf("vps rule: %s is already registered", name)
+		}
+	} else if cat == 2 {
+		if _, ok := vpsRulesVolume[name]; !ok {
+			vpsRulesVolume[name] = d
+		} else {
+			return fmt.Errorf("vps rule: %s is already registered", name)
+		}
+	} else if cat == 3 {
+		if _, ok := vpsRulesMix[name]; !ok {
+			vpsRulesMix[name] = d
+		} else {
+			return fmt.Errorf("vps rule: %s is already registered", name)
+		}
+	} else if cat == 4 {
+		if _, ok := vpsRulesMixScale[name]; !ok {
+			vpsRulesMixScale[name] = d
+		} else {
+			return fmt.Errorf("vps rule: %s is already registered", name)
+		}
+	} else if cat == 5 {
+		if _, ok := vpsRulesPending[name]; !ok {
+			vpsRulesPending[name] = d
+		} else {
+			return fmt.Errorf("vps rule: %s is already registered", name)
+		}
+	} else if cat == 6 {
+		if _, ok := vpsRulesDefaultLabel[name]; !ok {
+			vpsRulesDefaultLabel[name] = d
+		} else {
+			return fmt.Errorf("vps rule: %s is already registered", name)
+		}
+	} else {
+		return fmt.Errorf("vps rule category: %d, is not valid", cat)
+	}
+
+	return nil
+}
+
+// GetVpsRules return the list of vps rules
+func GetVpsRules(cat int) map[string]VpsTemplate {
+	if cat == 1 {
```

Comment:
instead of adding this put the rules in an indexed list like `[]map[string]VpsTemplate`, so you can use `return rules[cat]`

Reply:
Made the changes as per the suggestion